documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc

```diff
@@ -335,162 +335,115 @@ h|*27* h|*26* h|*25* h|*24* h|*23* h|*22* h|*21* h|*20* h|*19* h|*18* h|*17* h|*
 
 === Disable Other GPIO Peripherals
 
-Note that all other peripheral overlays that use conflicting GPIO pins must be disabled. In config.txt, take care to comment out or invert any dtparams that enable I2C or SPI:
+Note that all other peripheral overlays that use conflicting GPIO pins must be disabled. In `config.txt`, take care to comment out or invert any dtparams that enable I2C or SPI:
 
 ----
 dtparam=i2c_arm=off
 dtparam=spi=off
 ----
 
-=== Controlling Output Format
+=== Configure a Display
 
-The output format (clock, colour format, sync polarity, enable) can be controlled with a magic number (unsigned integer or hex value prefixed with 0x) passed to the `dpi_output_format` parameter in config.txt created from the following fields:
+The https://en.wikipedia.org/wiki/Direct_Rendering_Manager#Kernel_mode_setting[Kernel Mode Setting (KMS)] generic display interface enables output to arbitrary displays, as long as you have an appropriate driver.
+
+==== Auto Detect
+
+Auto detect allows your Raspberry Pi to connect with a display without a manually configured device tree overlay.
+Auto detection is enabled by default. You can enable display autodetect by adding the following line to `config.txt`:
 
 ----
-output_format          = (dpi_output_format >>  0) & 0xf;
-rgb_order              = (dpi_output_format >>  4) & 0xf;
-
-output_enable_mode     = (dpi_output_format >>  8) & 0x1;
-invert_pixel_clock     = (dpi_output_format >>  9) & 0x1;
-
-hsync_disable          = (dpi_output_format >> 12) & 0x1;
-vsync_disable          = (dpi_output_format >> 13) & 0x1;
-output_enable_disable  = (dpi_output_format >> 14) & 0x1;
-
-hsync_polarity         = (dpi_output_format >> 16) & 0x1;
-vsync_polarity         = (dpi_output_format >> 17) & 0x1;
-output_enable_polarity = (dpi_output_format >> 18) & 0x1;
-
-hsync_phase            = (dpi_output_format >> 20) & 0x1;
-vsync_phase            = (dpi_output_format >> 21) & 0x1;
-output_enable_phase    = (dpi_output_format >> 22) & 0x1;
-
-output_format:
-   1: DPI_OUTPUT_FORMAT_9BIT_666
-   2: DPI_OUTPUT_FORMAT_16BIT_565_CFG1
-   3: DPI_OUTPUT_FORMAT_16BIT_565_CFG2
-   4: DPI_OUTPUT_FORMAT_16BIT_565_CFG3
-   5: DPI_OUTPUT_FORMAT_18BIT_666_CFG1
-   6: DPI_OUTPUT_FORMAT_18BIT_666_CFG2
-   7: DPI_OUTPUT_FORMAT_24BIT_888
-
-rgb_order:
-   1: DPI_RGB_ORDER_RGB
-   2: DPI_RGB_ORDER_BGR
-   3: DPI_RGB_ORDER_GRB
-   4: DPI_RGB_ORDER_BRG
-
-output_enable_mode:
-   0: DPI_OUTPUT_ENABLE_MODE_DATA_VALID
-   1: DPI_OUTPUT_ENABLE_MODE_COMBINED_SYNCS
-
-invert_pixel_clock:
-   0: RGB Data changes on rising edge and is stable at falling edge
-   1: RGB Data changes on falling edge and is stable at rising edge.
-
-hsync/vsync/output_enable_polarity:
-   0: default for HDMI mode
-   1: inverted
-
-hsync/vsync/oe phases:
-   0: DPI_PHASE_POSEDGE
-   1: DPI_PHASE_NEGEDGE
+`display_auto_detect=1`
```

Comment:
This only applies to DSI displays, so might not be useful in the DPI docs?

Reply:
I might be misunderstanding, but I think DSI refers to a connector standard (cable on the Pi, as opposed to plugging in a microHDMI cable) and DPI (display pixel interface) refers to the communication standard, which in Bookworm now includes both HDMI and DSI. I think. I'm not sure it makes sense to split the two apart any more given that change.

Reply:
DSI = Display Serial Interface
DPI = Display Parallel Interface

So different signalling mechanisms, not just different types of connector.

Reply:
Wait! DPI = Display Pixel Interface surely?

https://www.itwissen.info/en/display-pixel-interface-DSI-DPI-128674.html#gsc.tab=0 (amongst other references?)

Reply:
I just asked @6by9 and he confidently said "Display Parallel Interface".
https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/raspberry-pi/display-parallel-interface.adoc
https://pinout.xyz/pinout/dpi

I guess it's just an unfortunate coincidence that "DPI" has two different expansions, depending on context :joy: :upside_down_face:

Reply:
And just to make it murkier still, the actual title of the spec is
> MIPI Alliance Standard for Display Pixel Interface (DPI-2)

Whilst the spec is at V2, I wouldn't say that the ```2``` on the end of ```DPI-``` refers to the version, in the same way as we implement CSI-2 v1.01 for cameras.

Reply:
I hate displays. I just want to make that really clear. 🤦‍♂️